Implemented functions and local storage

iKuartz opened the pull request:
For this part, what I did was:

    Add a new JavaScript file and import it as a module:
    It contains methods related to the status updates (completed: true/false).
    Add event listener to the checkbox (change).
    Update items object's value for completed key upon user actions.
    Implement a function for the "Clear all completed" button (use filter() method).
    Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

ivan-barbosa-da-silva committed:
initial commit

ivan-barbosa-da-silva committed:
Added add item function. No local storage yet.

ivan-barbosa-da-silva committed:
Add and check functions working properly. Still no local storage.

ivan-barbosa-da-silva committed:
Functions almost done, still getting one error

ivan-barbosa-da-silva committed:
functions implemented with local storage

ivan-barbosa-da-silva committed:
Removed console logs for debugging

ivan-barbosa-da-silva committed:
Linter corrections

ivan-barbosa-da-silva committed:
Linter corrections

ivan-barbosa-da-silva committed:
Linter corrections

ivan-barbosa-da-silva committed:
Linter corrections

ivan-barbosa-da-silva committed:
Linter corrections

ivan-barbosa-da-silva committed:
compiled code

elisha2kyakpo1 reviewed (changes requested):
Hi @iKuartz,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
- The PR description and title looks professional 👍


## Required Changes ♻️
_Check the comments under the review._

- Let's implement clear all completed function as per project instructions.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- The name of the project may mislead users, a Todo list can keep different items not just books(awesome books)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ivan-barbosa-da-silva committed:
Added new javascript structure, more organized, according to peer review. still needs testing

ivan-barbosa-da-silva committed:
Code fully tested, all functionalities implemented.

Meltrust commented on src/modules/delete.js:
```diff
@@ -2,10 +2,10 @@ export default function deleteTodo(array) {
   array = array.filter((obj) => obj.completed === false);
 
   for (let i = 0; i < array.length; i += 1) {
```
- [ ] OPTIONAL: It's a lot more efficient (faster) to use `map` or even `filter` instead of a regular for loop.  Try it! 💪

iKuartz replied:
Thank you! I will optimize it :)

Meltrust commented on src/modules/creation.js:
```diff
@@ -5,15 +5,21 @@ const list = document.querySelector('.todo-list');
+  const edits = document.getElementsByClassName('taskEdit');
 
+  /* eslint-disable no-plusplus */
   for (let i = 0; i < elements.length; i++) {
```
- [ ] OPTIONAL: It's a lot more efficient (faster) to use `map` or even `filter` instead of a regular for loop.  Try it! 💪

Meltrust commented on src/modules/add.js:
```diff
@@ -10,7 +10,7 @@ export default function addTodo(array) {
 
     // New element added to the beginning of the list
     array.push(todoObject);
```
- [ ] OPTIONAL: Favor the use array methods that don't mutate the array, e.g. [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push), [pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) and so on.

iKuartz replied:
Oh, ok! Thank you for the tip!

Meltrust reviewed (approved):
Hi @iKuartz,


## Wow, you did it 🎉
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for implementing the "clear all completed" button function, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

FrederickMih commented on src/modules/delete.js:
```diff
@@ -0,0 +1,11 @@
+export default function deleteTodo(array) {
+  array = array.filter((obj) => obj.completed === false);
+
+  for (let i = 0; i < array.length; i += 1) {
+    array[i].index = i + 1;
+  }
+
+  localStorage.setItem('todoList', JSON.stringify(array));
+
+  return array;
+}
```
- [x] Hi @iKuartz, a nice job for implementing a function to clear all the tasks in the collection. Please let's add another function for deleting a task(remove an element from the array). This way, you meet the project's requirements and get closer to approval.

Whoistolu replied:
I see this has been implemented and it works fine now 👍🏽

FrederickMih reviewed (changes requested):
#### Hi @iKuartz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![](https://media.giphy.com/media/L422JoNlBBBD42zkQ9/giphy.gif)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ivan-barbosa-da-silva committed:
Implemented the button to remove a single task

rukundoeric reviewed (changes requested):
Hi @iKuartz, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] I could not test the functionality of those features since I couldn't even be able to create a task due to an error that is occurring. check this [video](https://www.loom.com/share/ebe749a641d24426a14a7f92989baceb) to see what is happening. Kindly,  address the issue and fix it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

iKuartz commented:
> Hi @iKuartz,
> 
> _Great job so far clap 100 +1. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
> ## Highlights
> 
>     * Follows git workflow.  100
> 
>     * Codebase looks clean and professional. 100
> 
>     * Addressed some of the issues raised from the previous review. 100
> 
> 
> ## Required Changes recycle
> 
>     * [ ]  I could not test the functionality of those features since I couldn't even be able to create a task due to an error that is occurring. check this [video](https://www.loom.com/share/ebe749a641d24426a14a7f92989baceb) to see what is happening. Kindly,  address the issue and fix it.
> 
> 
> ## Optional suggestions
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
>     * N/A
> 
> 
> Cheers and Happy coding!clapclapclap
> 
> **Feel free to leave any questions or comments in the PR thread if something is not 100% clear.**
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> > As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Sorry, I have no idea how to address this issue in your machine. I will record a video showcasing the functionalities because they are definitely working on my end.

ivan-barbosa-da-silva committed:
Trying some corrections to make it work on GitHub

iKuartz commented:
This is a video of the present state of the website.
https://www.loom.com/share/51d7769ffc7a4f3b8826ca52584cbcfb

ivan-barbosa-da-silva committed:
next attempt

Whoistolu reviewed (approved):
### Hi @iKuartz,

Great job implementing the recommended changes, however, the issue the previous reviewer had was with webpack but the project works fine now.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
